Automattic/vip-go-mu-plugins
# Add a filter to check if the current user can purge the cache

## Description

This PR adds a new filter, `vip_cache_manager_can_purge_cache`, which controls whether the current user is allowed to purge the cache. Initially, this functionality was available only to users with the `manage_options` capability.

## Changelog Description

### Plugin Updated: Cache Manager

We added a filter to check if the current user can purge the cache.

`vip_cache_manager_can_purge_cache` returns a boolean whether current user should be able to see Flush Cache button. Defaults to `current_user_can( 'manage_options' )`.

```php
// Allow editors and up to be able to purge a page from cache
add_filter( 'vip_cache_manager_can_purge_cache', function( $can_purge, $user ) { 
	return current_user_can( 'edit_others_posts' );
}, 10, 2 );
```

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Check out PR.
2. Create a non-admin user and log in; observe that the user does not have an option to purge the page cache.
3. Assign the user `manage_options` capability. Observe that the user have an option and is actually able to purge the cache.
